airbyte-integrations/connectors/source-google-analytics-data-api/source_google_analytics_data_api/source.py

```diff
@@ -325,16 +325,20 @@ def check_connection(self, logger: logging.Logger, config: Mapping[str, Any]) ->
             if invalid_metrics:
                 invalid_metrics = ", ".join(invalid_metrics)
                 return False, f"custom_reports: invalid metric(s): {invalid_metrics} for the custom report: {report['name']}"
+            report_stream = self.instantiate_report_class(report, config)
+            # check if custom_report dimensions + metrics can be combined and report generated
+            stream_slice = next(report_stream.stream_slices(sync_mode=SyncMode.full_refresh))
+            next(report_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=stream_slice), None)
```

Comment:
In these we will need to except StopIterations, right? In case things are empty and either return no stream slices or no records for a given slice

Reply:
not really, as we'll always have al least 1 stream slice (for today)

Reply:
What about if for today there are no records?

Reply:
No problem, if we receive an empty `rows` response from Google, then no records will be produced. We check for a successful (200) response from Google.

Reply:
🤦🏻 I now see the default value you added here (the `, None). All good